lib/Parser/BGParseManager.cpp

```diff
@@ -94,6 +97,10 @@ BGParseWorkItem* BGParseManager::FindJob(DWORD dwCookie, bool waitForResults)
         if (item->GetCookie() == dwCookie)
         {
             matchedWorkitem = item;
```

Comment:
Should these loops break once they find a match? I know that there can be only one match because IDs are unique, so there's no functional difference, but maybe a `break` would add some clarity. I don't feel strongly on this, feel free to leave as-is if you prefer.

Reply:
the loops do 'break' with the for-loop condition (when matchedWorkItem != null).
i'll keep as is because I like to keep the invariant in one place

Reply:
Good point, I didn't notice that.